cmd/nerdctl/run_network_linux_test.go

```diff
@@ -140,6 +140,50 @@ func valuesOfMapStringString(m map[string]string) map[string]struct{} {
 	return res
 }
 
+func TestRunPortWithNoHostPort(t *testing.T) {
+	type testCase struct {
+		containerPort    string
+		runShouldSuccess bool
+	}
+	testCases := []testCase{
+		{
+			containerPort:    "8080",
+			runShouldSuccess: true,
+		},
+		{
+			containerPort:    "8080-8081",
+			runShouldSuccess: true,
+		},
+		{
+			containerPort:    "49153-61000",
+			runShouldSuccess: false,
+		},
+	}
+	tID := testutil.Identifier(t)
+	for i, tc := range testCases {
+		i := i
+		tc := tc
+		tcName := fmt.Sprintf("%+v", tc)
+		t.Run(tcName, func(t *testing.T) {
+			testContainerName := fmt.Sprintf("%s-%d", tID, i)
+			base := testutil.NewBase(t)
+			defer base.Cmd("rm", "-f", testContainerName).Run()
+			pFlag := tc.containerPort
+			cmd := base.Cmd("run", "-d",
+				"--name", testContainerName,
+				"-p", pFlag,
+				testutil.NginxAlpineImage)
+			if tc.runShouldSuccess {
+				cmd.AssertOK()
```

Comment:
This is not enough. We have to assure that the port is allocated and connectable.

Reply:
OK